Package/idealalphaPackage/R/main.R

```diff
@@ -0,0 +1,289 @@
+library(pwr)
+library(ggplot2)
+
+#functions to justify alpha levels.
+#first run all the code. In the beginning are functions incoroporated in later functions
+
+#call idealalpha(n, effsize, prior, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4)
+#to calculate the alpha level which gives optimal learning
+#the output will be alpha, power and info, which is the expected learning
+
+#call alphaConfirmation(n, effsize, prior, postfalse, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4)
+#to calculate the alpha level needed to achieve a given posterior after a significant result
+#the output will be alpha, power, info, and confirm, which is the probability after a positive result
+
+#alphaDisconfirmation(n, effsize, prior, postfalse, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4)
+#to calculate the alpha level needed to achieve a given posterior after a non-significant result
+#the output will be alpha power and disconfirm, which is the probability after a negative result
+
+#probpower(effsize, prior, postfalse, posttrue, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4)
+#conduct a probability based power analysis. Here posterior after significance and posterior after non-signficance are fixed
+#the function will then return the sample size and the alpha level needed
+#the output will be alpha, power, disconfirm, confirm, info and n, the sample size per group
+
+
+#The next functions are defined to be part of the four main functions
+
+#'Calculate the probability that the hypothesis is true, after a significant test result.
+#'@param power statistical power of the test.
+#'@param alpha alpha level.
+#'@param prior prior probability of H1.
+#'
+#'@examples postConfirm(0.8, 0.05, 0.5)
+#'@return posterior probability of hypothesis.
+#'@export
+postConfirm <- function(power,alpha,prior)
+                          {
+        (power*prior)/(power*prior+alpha*(1-prior))
+}
+
+#'Calculate the probability that the hypothesis is true, after a non-significant test result.
+#'@param power statistical power of the test.
+#'@param alpha alpha level.
+#'@param prior prior probability of H1.
+#'
+#'@examples postDisconfirm(0.8, 0.05, 0.5)
+#'@return posterior probability of hypothesis.
+#'@export
+postDisconfirm <- function(power, alpha, prior)
+{
+  (1-power)*prior/((1-alpha)*(1-prior)+(1-power)*prior)
+}
+
+#define power function for t.test, anova and test for correlations based on pwr package
+get_power   <- function(alpha, effsize, n, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4){ #get_power for different alpha levels, two tail independent samples t-test
```

Comment:
maybe you can split this line of code into two lines, it's hard to read on a normal laptop.